pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -208,20 +231,116 @@ func (r *ControlPlaneReconciler) Reconcile() (reconcile.Result, error) {
 	return reconcile.Result{Requeue: true}, nil
 }
 
+// mergeValues merges a map containing input values on top of a map containing
+// base values, giving preference to the base values for conflicts
+func mergeValues(base map[string]interface{}, input map[string]interface{}) map[string]interface{} {
+	if base == nil {
+		base = make(map[string]interface{}, 1)
+	}
+
+	for key, value := range input {
+		//if the key doesn't already exist, add it
+		if _, exists := base[key]; !exists {
+			base[key] = value
+			continue
+		}
+
+		// at this point, key exists in both input and base.
+		// If both are maps, recurse.
+		// If only input is a map, ignore it. We don't want to overrwrite base.
+		// If both are values, again, ignore it since we don't want to overrwrite base.
+		if baseKeyAsMap, baseOK := base[key].(map[string]interface{}); baseOK {
```

Comment:
Are values in the base config are supposed to be the default values? If so, shouldn't values in the input map take precedence?

Reply:
I think the naming is a bit misleading, but we're calling this with user-supplied config as `base` and the template config as `input`, so this behavior is correct.